hathor/manager.py

```diff
@@ -699,14 +825,52 @@ def on_new_tx(self, tx: BaseTransaction, *, conn: Optional[HathorProtocol] = Non
             # Propagate to our peers.
             self.connections.send_tx_to_peers(tx)
 
-        if self.wallet:
-            # TODO Remove it and use pubsub instead.
-            self.wallet.on_new_tx(tx)
+        return True
 
-        # Publish to pubsub manager the new tx accepted
+    def step_validations(self, txs: Iterable[BaseTransaction]) -> None:
+        """ Step all validations until none can be stepped anymore.
+        """
+        # cur_txs will be empty when there are no more new txs that reached full
+        # validation because of an initial trigger
+        for ready_tx in txs:
+            assert ready_tx.hash is not None
+            self.tx_storage.remove_ready_for_validation(ready_tx.hash)
+        for tx in map(self.tx_storage.get_transaction, self.tx_storage.next_ready_for_validation()):
+            assert tx.hash is not None
+            tx.update_initial_metadata()
+            try:
+                assert tx.validate_full()
+            except (AssertionError, HathorError):
+                # TODO
```

Comment:
This method is called from `on_new_tx` that has `fail_silently` parameter. Should we have the same here and use it in this exception?

Reply:
Makes sense, but rather than doing that I think it's better for on_new_tx to handle that exception and `return False` when there is one.

Reply:
I just did that, can you check it out?

Reply:
I agree is better the way you did